### ISSUE TITLE
Trying to create SF cluster using azure CLI fails.

### ISSUE THREAD

**Issue by 01011011** [Azure/azure-cli#12470]
## Describe the bug

**Command Name**
`az sf cluster create`

**Errors:**
```
'NoneType' object has no attribute 'get'
Traceback (most recent call last):
python3.6/site-packages/knack/cli.py, ln 206, in invoke
    cmd_result = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 608, in execute
    raise ex
cli/core/commands/__init__.py, ln 666, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
...
cli/command_modules/servicefabric/custom.py, ln 1158, in _create_certificate
    cmd, cli_ctx, vault_resource_group_name, vault_name, location, enabled_for_deployment=True).result()
cli/command_modules/servicefabric/custom.py, ln 1667, in _create_keyvault
    permissions = Permissions(keys=[KeyPermissions.get,
AttributeError: 'NoneType' object has no attribute 'get'
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az sf cluster create -g {} -c {} -l {} --cluster-size {} --vm-password {} --certificate-output-folder {} --certificate-subject-name {}`

## Environment Summary
```
Linux-4.15.0-1071-azure-x86_64-with-debian-stretch-sid
Python 3.6.5

azure-cli 2.1.0

```
## Additional Context

![image](https://user-images.githubusercontent.com/20220724/76035102-7e77cb80-5ef5-11ea-89d3-1e32a639e1d9.png)

**Comment by yungezz** [comment]
thanks for reporting, we're looking at it.

**Comment by a-santamaria** [comment]
Thanks for reporting. this is caused by a breaking change introduced in the new key vault management sdk (azure-mgmt-keyvault) introduced in azure-cli 2.1.0. Unfortunately we didn't have test coverage for this path, I'll send a pr shortly with the fix and updating the tests. Please use previews version [Azure CLI 2.0.81](https://github.com/Azure/azure-cli/releases/tag/azure-cli-2.0.81) in the meantime

**Comment by 01011011** [comment]
Thanks Alfredo.  I will check it out.

**Comment by rameshgit001** [comment]
My reported issue as in #12811 (which was closed) is still occurring as of today. Is the fix to this problem still pending deployment to production? Thanks.

**Comment by 01011011** [comment]
you can use preview CLI 2.0.81 in the meantime as noted. I dont believe this is in prod yet

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by a-santamaria** [comment]
The fix is released in CLI version 2.4.0
